presto-main/src/main/java/com/facebook/presto/memory/MemoryManagerConfig.java

```diff
@@ -72,4 +73,11 @@ public MemoryManagerConfig setMaxQueryMemory(DataSize maxQueryMemory)
         this.maxQueryMemory = maxQueryMemory;
         return this;
     }
+
+    public static class LowMemoryKillerPolicy
```

Comment:
I don't think we need multiple policies. We don't use the existing one, so no point in keeping it around.

Reply:
Maybe there are external users that use it?

I agree it's best if a single policy / heuristic can work for everything, though we should consider if there are different types of cluster configurations and workloads that would require different policies (or config for the policy).

Reply:
The purpose of the killer is to unblock the cluster. The current policy doesn't solve this problem because of how queries get blocked (insufficient memory at a local level) and lack of memory-awareness when scheduling workload that lead to skew.

Reply:
Said another way, if they are using it, it probably doesn't do what they think or hope it's doing.

Also, this adds unnecessary complexity to the code (extra classes/interfaces/structure).

Reply:
The policy in this pull request is an attempt to improve the existing policy. It has a few properties that are more desirable than the existing one, but I don't think it's perfect. In fact, it's a compromise from our original proposal (which wasn't implemented here because of its undesirable behavior in edge cases).

I feel like we will iterate on this, and making this flexible is useful.

Reply:
Seems like premature over-engineering to me. If and when we need to iterate (in ways other than modifying the existing code), we can add the flexibility then. Until that time, it's just hypothetical.

Reply:
I’m fine with removing the old policy